### PR TITLE
feat: add melee monster

### PR DESCRIPTION
**Pull request by JoenvandeVorle**
## Description

Adds the melee enemy prefab. It can attack and kill the player, with a delay between start of the attack and the hit. 
Also includes some basic player logic so that the player can die.

## Testability

Open the MeleeEnemy scene. A blue respawning enemy should be there and start attacking the player. Once either your or the enemies' health reaches zero you respawn. The enemy grows in size when attacking as a visual indicator. 
There is also a `MeleeEnemyTest` aplib test

## Checklist
- [x] Set the proper pull request name
- [x] Merged main into your branch
- [x] Added a scene to the game for your changes

### CODE REVIEW

File: aplib.net-demo/Assets/Scripts/Entities/HealthComponent.cs [hunk 1]
```diff
@@ -33,6 +33,11 @@ public class HealthComponent : MonoBehaviour
     /// </summary>
     public int Health { get; private set; }
 
+    /// <summary>
+    /// The maximum health of the GameObject.
+    /// </summary>
+    public int MaxHealth {get => _maxHealth;}
```
Comment (Tboefijn) [inline aplib.net-demo/Assets/Scripts/Entities/HealthComponent.cs:39]:
```suggestion
    public int MaxHealth { get => _maxHealth; }
```

**Review by Tboefijn — CHANGES_REQUESTED** [review]
LGTM! Left a styling comment.

**Review by joachimdekker — DISMISSED** [review]
LGTM!

**Review by Tboefijn — APPROVED** [review]
LGTM!

**Review by joachimdekker — APPROVED** [review]

**Pull request merged**